Tools/llm-tool/LLMTool.swift

```diff
@@ -203,29 +206,88 @@ struct EvaluateCommand: AsyncParsableCommand {
 
     @MainActor
     mutating func run() async throws {
-        let (modelContainer, modelConfiguration) = try await memory.start(args.load)
+        let modelContainer = try await memory.start { [args] in
+            try await args.load(
+                defaultModel: "mlx-community/Mistral-7B-v0.1-hf-4bit-mlx",
+                modelFactory: LLMModelFactory.shared)
+        }
+        let modelConfiguration = modelContainer.configuration
 
         if !generate.quiet {
             print("Model loaded -> \(modelConfiguration.id)")
         }
 
         let prompt = generate.prompt ?? modelConfiguration.defaultPrompt
-        let messages = [["role": "user", "content": prompt]]
-        let promptTokens = try await modelContainer.perform { _, tokenizer in
-            try tokenizer.applyChatTemplate(messages: messages)
-        }
 
         if !generate.quiet {
             print("Starting generation ...")
             print(prompt, terminator: "")
         }
 
-        let result = await modelContainer.perform { [generate] model, tokenizer in
-            generate.generate(
-                promptTokens: promptTokens, model: model, tokenizer: tokenizer,
-                extraEOSTokens: modelConfiguration.extraEOSTokens)
+        let result = try await modelContainer.perform { [generate] context in
+            let input = try await context.processor.prepare(input: .init(prompt: prompt))
```

Comment:
Sorry this might be a dumb newbie question but why do you not capture the `prompt` explicitly here but you do capture the input in the VLM example?

Reply:
`prompt` is a `let` (constant) value, while `input` (in the llm case) is `var` so it needs to be captured explicitly to signal we want to send it across the isolation context.  If `input` was also `let` it wouldn't be needed.